logstash-core/lib/logstash/bootstrap_check/default_config.rb

```diff
@@ -0,0 +1,17 @@
+# encoding: utf-8
+require "logstash/errors"
+
+module LogStash module BootstrapCheck
+  class DefaultConfig
+    def self.check(settings)
+      if settings.get("config.string").nil? && settings.get("path.config").nil?
+        raise LogStash::BootstrapCheckError, I18n.t("logstash.runner.missing-configuration")
```

Comment:
I think using `pipeline.yml` is an overkill, IMO. There should be one file to configure Logstash. In production, users would just use the file source(s). I understand the debate is about reloadable configurations from the settings file, but that is out of scope for this PR.